cli/tools/repl/editor.rs

```diff
@@ -428,6 +437,21 @@ impl ReplEditor {
   }
 }
 
+/// Command to reverse search history , same as rustyline default C-R but that resets repl should_exit flag to false
+struct ReverseSearchHistoryEventHandler(Arc<AtomicBool>);
```

Comment:
Since the field is unrelated to the name of the struct, it would be good to give it a name to make this code a little more clear in isolation:

```rs
struct ReverseSearchHistoryEventHandler {
  should_exit_on_interrupt: Arc<AtomicBool>,
}
```